src/Symfony/Component/DependencyInjection/Dumper/YamlDumper.php

```diff
@@ -306,6 +306,8 @@ private function dumpValue($value)
             return new TaggedValue('service', (new Parser())->parse("_:\n".$this->addService('_', $value), Yaml::PARSE_CUSTOM_TAGS)['_']['_']);
         } elseif ($value instanceof AbstractArgument) {
             return new TaggedValue('abstract', $value->getText());
+        } elseif ($value instanceof \UnitEnum) {
+            return new TaggedValue('php/enum', $value);
```

Comment:
the value passed to the TaggedValue does not match what you expect in the loader.